tephra-hbase-compat-0.96/src/test/java/co/cask/tephra/hbase96/coprocessor/TransactionProcessorTest.java

```diff
@@ -264,7 +268,7 @@ public void testDeleteMarkerCleanup() throws Exception {
       region.initialize();
 
       // all puts use a timestamp before the tx snapshot's visibility upper bound, making them eligible for removal
-      long writeTs = txSnapshot.getVisibilityUpperBound() - 10;
+      long writeTs = ((TransactionSnapshot) txVisibilityState).getVisibilityUpperBound() - 10;
```

Comment:
Since `getVisibilityUpperBound()` deals with visibility, we should add it to `TransactionVisibilityState` interface. Then we wouldn't need this cast.